src/pages/Dashboard/DashboardSidebar.tsx

```diff
@@ -59,13 +60,30 @@ export const DashboardSidebar = memo(() => {
   const {
     state: { isConnected },
   } = useWallet()
+
+  const [isMounted, setIsMounted] = useState(false)
+  const shouldRenderRecentTransactions = useDeferredValue(isMounted)
+
+  useEffect(() => {
+    // Seems useless, but ensures this is set on next tick (which isn't available on browsers)
+    const timerId = setTimeout(() => {
+      setIsMounted(true)
+    }, 0)
+
+    return () => clearTimeout(timerId)
+  }, [])
+
   if (!isConnected) return null
 
   return (
     <Flex width='full' flexDir='column' gap={6}>
       <PromoCard data={promoData} />
       <EligibleCarousel />
-      <RecentTransactions limit={8} viewMoreLink display={displayXlFlex} />
+      {shouldRenderRecentTransactions ? (
+        <RecentTransactions limit={8} viewMoreLink display={displayXlFlex} />
+      ) : (
+        <RecentTransactionsSkeleton limit={5} viewMoreLink display={displayXlFlex} />
+      )}
```

Comment:
q: should we use suspense?

Reply:
Not yet, as I found out `Suspense` does nothing in this PR for the time being, since we're doing the deferred dance on the parent (haven't really found a way to make it work on the children)

Removed in 6364b78f759e3b9aa8b97fa336315a51ccb25ece